packages/bpk-component-autosuggestV2/src/BpkAutosuggest.tsx

```diff
@@ -0,0 +1,406 @@
+  sectionTitle?: string;
+  clearButton?: string;
+  label?: string;
+  desktopLabel?: string;
+  input?: string;
+  inputWrapper?: string;
+  inputTextWrapper?: string;
+  desktopSuggestionsContainer?: string;
+  desktopSuggestionsList?: string;
+};
+
+export type BpkAutoSuggestProps<T> = {
+  ariaLabels: {
+    resultsList?: string;
+    label?: string;
+    clearButton?: string;
+  };
+  getSuggestionValue: (suggestion: T) => string;
+  id: string;
+  inputProps: HTMLProps<HTMLInputElement>;
+  onSuggestionsClearRequested: () => void;
+  onSuggestionSelected?: (data?: {
+    inputValue: string;
+    suggestion?: T;
+  }) => void;
+  onSuggestionsFetchRequested: (value: string) => void;
+  renderSuggestion: (suggestion: T) => ReactElement;
+  suggestions: T[];
+  getA11yResultsMessage: (resultCount: number) => string;
+  defaultValue?: string;
+  enterKeyHint?: string;
+  isBanana?: boolean;
+  isDesktop?: boolean;
+  onClick?: () => void;
+  onLoad?: (inputValue: string) => void;
+  renderBesideInput?: () => ReactElement;
+  showClear?: boolean;
+  theme?: Partial<BpkAutoSuggestTheme>;
+  highlightFirstSuggestion?: boolean;
+  shouldRenderSuggestions?: (value?: string) => boolean;
+  multiSection?: boolean;
+  getSectionSuggestions?: (section: T) => T[];
+  renderSectionTitle?: (section: T) => ReactElement;
+  alwaysRenderSuggestions?: boolean;
+  onInputValueChange?: (input: { method: string; newValue: string }) => void;
+};
+
+const defaultTheme = {
+  container: getClassName('bpk-autosuggest__container'),
+  containerOpen: getClassName('bpk-autosuggest__container--open'),
+  suggestionsContainer: getClassName('bpk-autosuggest__suggestions-container'),
+  suggestionsContainerOpen: getClassName(
+    'bpk-autosuggest__suggestions-container--open',
+  ),
+  suggestionsList: getClassName('bpk-autosuggest__suggestions-list'),
+  suggestion: getClassName('bpk-autosuggest__suggestion-item'),
+  suggestionHighlighted: getClassName(
+    'bpk-autosuggest__suggestion-item--highlighted',
+  ),
+  sectionContainer: getClassName('bpk-autosuggest__section-container'),
+  sectionTitle: getClassName('bpk-autosuggest__section-title'),
+};
+
+const BpkAutosuggest = forwardRef<HTMLInputElement, BpkAutoSuggestProps<any>>(
+  (
+    {
+      alwaysRenderSuggestions,
+      ariaLabels,
+      defaultValue,
+      enterKeyHint,
+      getA11yResultsMessage,
+      getSectionSuggestions,
+      getSuggestionValue,
+      highlightFirstSuggestion,
+      id,
+      inputProps,
+      isBanana,
+      isDesktop,
+      multiSection,
+      onClick,
+      onInputValueChange,
+      onLoad,
+      onSuggestionSelected,
+      onSuggestionsClearRequested,
+      onSuggestionsFetchRequested,
+      renderBesideInput,
+      renderSectionTitle,
+      renderSuggestion,
+      shouldRenderSuggestions,
+      showClear,
+      suggestions,
+      theme: customTheme,
+    },
+    forwardedRef,
+  ) => {
+    const theme = { ...defaultTheme, ...customTheme };
```

Comment:
We need this for bananas custom theming but also to support [these use cases](https://github.com/Skyscanner/backpack/pull/2854#discussion_r1210310436)

Reply:
We shouldn't need any custom theming for banana (or any other project) unless any BPK designer has a agreed to it beforehand. Supporting custom theming like this would open too many possibilities for consumers to override the BPK design (with the assumption that we currently do not support this in the autosuggest component)
We do have custom theming in some components, but that was for different use cases i.e. partner branding and brand refresh

Reply:
That's interesting as the focus ring for me is blue, I think it depends on the browser?
Anyways, the idea would be that we should align the designs and not need a custom design for banana. As I said, it's ok to have different designs between mobile and desktop, but not between banana or any other specific project and bpk.
So we should probably be going back to the BPK designers and your designer to make sure the designs are aligned

Reply:
>focus ring is a different colour with larger border radius

This is something we are already working on it Backpack to add better focus indicators to our components across the business as part of A11y

>label lives inside input container

It's painful that currently Adam the Backpack designer is on vacation right now as I rememeber there are conversations ongoing about the inputs right now and how they should be.

Why does this have to be through a theme rather than configuration to the `BpkInput` component that can then be passed through.

>clear button is bigger

Similar to the above this doesn't align with our current components in Figma for the Backpack library so we need to check and sort this in design before we can just drop this into the component

Reply:
> Why does this have to be through a theme rather than configuration to the BpkInput component that can then be passed through.

The alternative is banana passing through its own element to render in place of the default BpkInput. It was found it was quite messy as it was reliant on useCombobox attributes being passed back out to banana. The theme and isBanana props felt the cleanest way.

I can go back to the drawing board and try something different though.